nucypher/policy/conditions/context.py

```diff
@@ -116,20 +118,31 @@ def get_context_value(context_variable: str, **context) -> Any:
     return value
 
 
-def _resolve_context_variable(param: Union[Any, List[Any]], **context):
+def resolve_any_context_variables(
+    param: Union[Any, List[Any], Dict[Any, Any]], **context
+):
     if isinstance(param, list):
-        return [_resolve_context_variable(item, **context) for item in param]
-    elif is_context_variable(param):
-        return get_context_value(context_variable=param, **context)
+        return [resolve_any_context_variables(item, **context) for item in param]
+    elif isinstance(param, dict):
+        result = {}
+        for k, v in param.items():
+            result[k] = resolve_any_context_variables(v, **context)
+        return result
+    elif isinstance(param, str):
+        # either it is a context variable OR contains a context variable within it
+        # TODO separating the two cases for now out of concern of regex searching
+        #  within strings (case 2)
+        if is_context_variable(param):
+            return get_context_value(context_variable=param, **context)
+        else:
+            matches = re.findall(CONTEXT_REGEX, param)
+            for context_var in matches:
+                # checking out of concern for faulty regex search within string
+                if context_var in context:
+                    resolved_var = get_context_value(
+                        context_variable=context_var, **context
+                    )
+                    param = param.replace(context_var, str(resolved_var))
```

Comment:
This is very defensive - don't fully trust your regex skill? :-)

Reply:
Haha that and I'm worried about legit `:<>` expressions within a string field.

Not applicable because I think it is not allowed for JsonPath, but a ridiculous example being some expression syntax analogous to the JsonPath query using a legit ":<>" combination.